### PR TITLE
[AIRFLOW-3639] Fix request creation in Jenkins Operator

### PR DESCRIPTION
**Pull request by vapiravfif**
### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3639

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Change Jenkins Operator to work with native Jenkins library method to configure REST request headers correctly

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Uses original operator's tests

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by vapiravfif** [comment]
Hi,

The failed tests doesn't seem to be related to the commit...

16475.2 and 16475.5 failed on:
```
[fail] 0.06% tests.www_rbac.test_views.TestTriggerDag.test_trigger_dag_button: [32m0.9072s[0m
```

and 16475.8 failed on:
```
[error] 0.01% tests.contrib.minikube.test_kubernetes_executor.KubernetesExecutorTest.test_integration_run_dag: [32m0.0075s[0m
[error] 0.00% tests.contrib.minikube.test_kubernetes_executor.KubernetesExecutorTest.test_integration_run_dag_with_scheduler_failure: [32m0.0035s[0m
```

**Comment by OmerJog** [comment]
mmm PR https://github.com/apache/airflow/pull/4414  is requesting to remove `jenkins_request_with_headers`  function so if #4414 is going to be merged this PR is not needed (?).
I'm not familiar with Jenkis just getting it to the attention of committers.

**Comment by vapiravfif** [comment]
PR #4414 rewrites whole Jenkins operator in Airflow, and is, actually, better.
But there is no activity there, and this PR fixes in minimalistic way the authentication problem that is a blocker for using Jenkins operator. If #4414 gets approved - this one indeed is not needed.
But I don't see why not to merge this, and when the #4414 is sorted out - it will replace the whole thing.

File: airflow/contrib/operators/jenkins_job_trigger_operator.py [hunk 1]
```diff
@@ -35,27 +35,21 @@
     basestring = str  # For python3 compatibility
 
 
-# TODO Use jenkins_urlopen instead when it will be available
-# in the stable python-jenkins version (> 0.4.15)
```
Comment (ashb) [inline airflow/contrib/operators/jenkins_job_trigger_operator.py:39]: Do you know what this comment was about? Do we need to update the version we specify in setup.py for `jenkins_request()` method to be available?

Comment (ashb) [reply airflow/contrib/operators/jenkins_job_trigger_operator.py:39]: I think this was added in 1.0.0.

Comment (vapiravfif) [reply airflow/contrib/operators/jenkins_job_trigger_operator.py:39]: Seems that jenkins_request was added in python-jenkins 1.0.0 
That is also the version where they changed the authentication mechanism that causes original operator not to work. Authentication no more initialized on connection creation, there is an internal method for that now.
1.0.0 - https://git.openstack.org/cgit/openstack/python-jenkins/tree/jenkins/__init__.py?h=1.0.0
0.4.16 (single release between 0.4.15 and 1.0.0) - https://git.openstack.org/cgit/openstack/python-jenkins/tree/jenkins/__init__.py?h=0.4.16

Comment (vapiravfif) [reply airflow/contrib/operators/jenkins_job_trigger_operator.py:39]: @ashb Added a commit to update setup.py

**Pull request merged**